aesd-char-driver/main.c

```diff
@@ -186,14 +191,94 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
     mutex_unlock(&(buffer->lock));
     return retval;
 }
+// ------------------------------aesd_adjust_file_offset-------------------------------------------
+
+static long aesd_adjust_file_offset(struct file *filp, unsigned int write_cmd, unsigned int write_cmd_offset)
+{
+    int i, tempfilepos, retval = 0;
+
+    struct aesd_dev *buffer = filp->private_data;
+
+    mutex_lock(&aesd_device.lock);
```

Comment:
@neohere97 I'd suggest using the interruptible mutex locking unless you have a specific reason you can't.  That way you can break out with Ctrl->C or signals.